configure.ac

```diff
@@ -61,8 +61,8 @@ case $host in
      lt_cv_deplibs_check_method="pass_all"
   ;;
 esac
-dnl Require C++11 compiler (no GNU extensions)
-AX_CXX_COMPILE_STDCXX([11], [noext], [mandatory], [nodefault])
+dnl Require C++17ish compiler (no GNU extensions)
+AX_CXX_COMPILE_STDCXX([17], [noext], [optional], [nodefault])
```

Comment:
We cannot make this optional if we want to rely on C++17 features (which we do, specifically `std:: optional` and `std:: variant`in order to remove Boost usage from `libzcashconsensus`).

Reply:
The issue is that the macro is very strict about what counts as C++17 support; it basically has a test suite of things that are supposed to compile under C++17. If there's anything missing in the support, even that we don't use, *or* if there is a bug in the macro that causes support not to be recognized, that causes a problem. If it turns out that this can be made mandatory again (possibly with some modifications to the macro) then I'll make it so, but it's easier to have it this way for experimentation.